lib/ui/pages/qrscanner_page.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:lottie/lottie.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:qrapp/services/qrservice.dart';
+import 'package:qrapp/ui/components/qrstatuswidget.dart';
+
+class QRScannerPage extends StatefulWidget {
+  const QRScannerPage({Key? key}) : super(key: key);
+
+  @override
+  _QRScannerPageState createState() => _QRScannerPageState();
+}
+
+class _QRScannerPageState extends State<QRScannerPage> {
+  final GlobalKey qrScannerKey = GlobalKey(debugLabel: 'QR Scanner');
+  late QRService provider;
+  Barcode? qrResults;
+  QRViewController? qrController;
+
+  bool scanningComplete = false;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() {
+      provider = Provider.of<QRService>(context, listen: false);
+    });
```

Comment:
**question:** why does this need to be wrapped in a setState call?

Reply:
Also you could directly use `late final` and initialize the value directly. 

`setState` is used to mark the `Element` behind the widget as "dirty" to trigger a rebuild on the next frame. You don't need it in `initState` since the Element and the RenderObject haven't been created yet at this time so you'll get a correct setup anyway!